### ISSUE TITLE
macOS - mouse isn't fully captured when cclosing the console

### ISSUE THREAD

**Issue by BielBdeLuna** [RobertBeckebans/RBDOOM-3-BFG#584]
Whenever the console is pulled down the mouse should be released from the window, 
In Linux ( Ubuntu ) recently the mouse has stopped being released. in order to release the mouse right now you have to alt-tab.

**Comment by SRSaunders** [comment]
I made this change in commit# 0aae6f0902cc25cb50a288e5a1fe8dd5dbb0da3a to prevent the cursor from being released during normal gameplay when the console is opened.  This is the same behaviour as when escape is hit during a game to open the game menu - it is not released then either.  Note that at the main menu the mouse is always released, as well as during demo playback when the console is opened.  And as you said above, alt-tab will also release the mouse when in a game (as well as pausing the game).

Previously, if the mouse was out of frame when closing the console during a game, the window boundary was not respected and the mouse could move out of frame during gameplay, at least on macOS.  I am not sure if this can be duplicated on other platforms, but it is possible and likely depends on the SDL2 implementation.  This lead to some confusing behaviour for the game player.

I felt this change was a reasonable gameplay safeguard, and alt-tab was a reasonable workaround.

The code change was in common_frame.cpp:
```
// SRS - don't release when console open in a game (otherwise may be out of frame on return) but always release at main menu
if( ( console->Active() && !game ) || !mapSpawned || ImGuiTools::ReleaseMouseForTools() )_
```
The previous code was:
```
if( console->Active() || ImGuiTools::ReleaseMouseForTools() )
```

**Comment by BielBdeLuna** [comment]
when you're mapping this becomes a problem, maybe we could have the old behaviour when in developer mode?

**Comment by SRSaunders** [comment]
If this is blocking an important workflow, I have no problem reverting the change.  I could always look deeper into what is causing the issue on macOS after re-grabbing the mouse when out of frame on console close.  Does linux have this issue?

I think Robert is getting close to his planned 1.3 release.  Do you want to submit a pull request or me?

**Comment by BielBdeLuna** [comment]
in Linux, before your commit, when the console was pulled down or the menu was on ( I can't confirm this on RBDoom3BFG before your commit, in OTE though I don't have any menu active right now, so I can't truly confirm it ), that is when not running a map, the mouse was released from the game window, when the mouse is on the game window the Operating System cursor is not visible, only becoming visible when the mouse is outside the game window.
Now once running a map, and once the console is pulled up, therefore not active any more, if the mouse is not over the game window, the mouse is still not captured, and the Operating System cursor is visible, it's only when the mouse enters the game window that the mouse is truly captured, and the player rotation is applied, until we pull down the console again at least.

once the mouse is captured by the game, the mouse doesn't exit the game window, and the Operating System cursor is not visible.
I see that doesn't happen in macOS, isn't it?

**Comment by SRSaunders** [comment]
I tested this on Manjaro linux and the mouse capture behaves properly.  As you say, when not running a map and at the main menu, the mouse is free to move on and off the window independent of whether the console is open or not.  When in a game and running a map, the mouse can be freed by opening the console (with my new pull request above).  If the mouse is not over the game window when the console is closed, the mouse is automatically moved back into the game window and is captured for use by the game and cannot leave the game window until the console is opened again or the until the game exits back to the main menu.  This is expected and good behaviour.

With macOS, everything works correctly except for the case when the mouse is outside of an active game window (i.e. running a map) when the console is closed.  In this case, the mouse is automatically moved back into the window but does not seem to be fully captured.  I can move the mouse outside of the game window without reopening the console, even with the game running an active map.  At that point, the only way to have the game fully capture the mouse is to click once outside of the game window (i.e. put the window into the background), and then click back into the game window bringing it back into the foreground.  At that point the mouse is fully captured and cannot leave the game window until the console is opened again or the until the game exits back to the main menu.  There is something different about the (SDL?) mouse control implementation on macOS vs linux.  I will have to investigate this further to hopefully track down the issue.

In the mean time I submitted a pull request to revert my original change, and the mouse will now be released with the console.  Hopefully this addresses your concern about map development workflow.
